fiftyone/core/aggregations.py

```diff
@@ -631,28 +700,41 @@ def to_mongo(self, sample_collection):
         return pipeline
 
     def _parse_args(self):
-        if self.bins is None:
+        if self._bins is None:
             bins = 10
         else:
-            bins = self.bins
+            bins = self._bins
+
+        if self._auto:
+            if etau.is_numeric(bins):
+                self._num_bins = bins
+            else:
+                self._num_bins = 10
 
-        if etau.is_numeric(bins):
-            if self.range is None:
-                # Automatic bins
-                self._bins = bins
-                return
+            return
+
+        if not etau.is_numeric(bins):
+            # User-provided bin edges
+            self._edges = list(bins)
+            return
 
+        if self._range is not None:
             # Linearly-spaced bins within `range`
             self._edges = list(
-                np.linspace(self.range[0], self.range[1], bins + 1)
+                np.linspace(self._range[0], self._range[1], bins + 1)
             )
-            return
+        else:
+            # Compute bin edges from bounds
+            self._num_bins = bins
 
-        # User-provided bin edges
-        self._edges = list(bins)
+    def _compute_bin_edges(self, sample_collection):
```

Comment:
This is convenient, although it means if `n` `HistogramValues` aggregations are executed via `aggregate([...])`, then `n +1` queries will be executed. I was hoping to keep the invariant that one (or a constant) number of queries are executed when using the `aggregate()` method.

Reply:
yeah I agree, but my thinking was that `$bucketAuto` is the only fully automatic option, and it makes all histograms look like the uniform distribution, which is actively harmful for CV/ML applications (one wants to see the shape)... so I thought that shouldn't be the default. I think forcing the user to provide the range is the only other option, but that feels more painful to the user in most cases than just slowing down their query a touch